src/libcore/iter.rs

```diff
@@ -3270,6 +3270,37 @@ impl<A, B> DoubleEndedIterator for Zip<A, B> where
 ///
 /// [`map()`]: trait.Iterator.html#method.map
 /// [`Iterator`]: trait.Iterator.html
+///
+/// # Notes about side effects
+///
+/// The [`map()`] iterator implements [`DoubleEndedIterator`], meaning that
+/// you can also [`map()`] backwards:
+///
+/// ```rust
+/// let v: Vec<i32> = vec![1, 2, 3].into_iter().map(|x| x + 1).collect();
+///
+/// assert_eq!(v, [2, 3, 4]);
+/// ```
+///
+/// [`DoubleEndedIterator`]: trait.DoubleEndedIterator.html
+///
+/// However, if you call `.rev()` to iterate in the other direction, it
+/// will only be 'backwards' if your closure has no state. In other words,
+/// something like this might act in a way that you wouldn't expect:
+///
+/// ```rust
+/// let counter = vec![1, 2, 3].into_iter().map({
+///     let mut ctr = 0;
+///     move |_| {
+///        ctr += 1;
+///         ctr
+///     }
+/// });
+///
+/// let counter2 = counter.reverse();
```

Comment:
I think I would find the example clearer if you included the items being iterated over, and they were not the numbers 1, 2, and 3. For example:

```
vec!['a', 'b', 'c'].into_iter().map(...).rev();
```

will produce `('c', 1), ('b', 2), ('a', 3)`, right?